### PR TITLE
[YouTube] Add support for attributed text description

### PR DESCRIPTION
**Pull request by Theta-Dev**
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [X] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- no API changes

YouTube is currently A/B testing a new video description format. The new format includes chips with the YouTube icon for internal links, as shown below:

image

Internally the new format does not use a list of text components but a string with the entire plaintext content along with a list of `commandRuns` which include the link data. I included a sample file for testing (`attributed_description.json`).

Because NewPipeExtractor falls back to the description extracted from the player endpoint, being unable to extract the new description format does not cause any issues when using NewPipe. However this should fix the description unit tests.

### CODE REVIEW

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java [hunk 1]
```diff
@@ -48,4 +54,12 @@ public void testConvertFromGoogleCacheUrl() {
...
+    }
```
Comment (FireMasterK) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java:64]: Why not just update the mocks instead?

**Review by FireMasterK — APPROVED** [review]

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java [hunk 2]
```diff
@@ -1,8 +1,5 @@
 package org.schabi.newpipe.extractor.services.youtube;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-
```
Comment (AudricV) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java:5]: Could you revert this?

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java [hunk 3]
```diff
@@ -12,6 +9,9 @@
 
 import java.io.IOException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
Comment (AudricV) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelperTest.java:14]: Same as above.

**Review by AudricV — CHANGES_REQUESTED** [review]
Thanks, please revert the test changes which are unrelated to your PR (Bandcamp + PeerTube + Learning playlist) and address the comments above.

**Comment by AudricV** [comment]
I resolved the conflicts, rebased the PR, removed the test changes (please open a separate PR for that), improved a bit the documentation of the method you added and squashed all the commits, as they were redundant.

Thanks for your hard work!

**Review by AudricV — APPROVED** [review]

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java [hunk 4]
```diff
@@ -1009,6 +1009,90 @@ public static String getTextFromObject(final JsonObject textObject, final boolea
...
+    @Nullable
+    public static String getAttributedDescription(
+            @Nullable final JsonObject attributedDescription) {
+        if (isNullOrEmpty(attributedDescription)) {
+            return null;
+        }
+
+        final String content = attributedDescription.getString("content");
+        final JsonArray commandRuns = attributedDescription.getArray("commandRuns");
+        if (content == null || commandRuns.isEmpty()) {
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:1032]: Isn't it ok for commandRuns to be empty?

**Review by Stypox — CHANGES_REQUESTED** [review]

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java (hunk 4 above)
Comment (FireMasterK) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:1032]: We should just return `content` in such cases imo

Comment (Theta-Dev) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:1032]: Yes, this check can be omitted as the remaining text is always appended, even if threre are no command runs

**Pull request merged**